textile/features.textile

```diff
@@ -485,29 +483,37 @@ h3(#realtime-connection). Connection
-** @(RTN16d)@ When a connection is successfully recovered, the @Connection#id@ will be identical to the @id@ of the connection that was recovered, and @Connection#key@ will always be updated to the @ConnectionDetails#connectionKey@ provided in the first @CONNECTED@ @ProtocolMessage@
-** @(RTN16e)@ If the @recover@ option is missing or no longer valid when connecting to Ably, the client will connect anyway, but emit a @ConnectionStateChange@ with a @reason@, and will additionally set the @Connection#errorReason@ with an @ErrorInfo@ object describing the failure
-** @(RTN16f)@ The @msgSerial@ component of the @recoveryKey@, unlike the other two components, is not sent to Ably, but rather is used to set the library internal @msgSerial@. (If the recover fails, the counter should be reset to 0 per "RTN15c3":#RTN15c3 )
+** @(RTN16i)@ Connection recovery is similar to connection resumption (see "RTN15c":#RTN15c), except that instead of the library resuming from a time at which that library instance was previously connected, it is doing so from external state provided in the client options, "(TO3i)":#TO3i. Since the library has no state at the time of connection, the channels must be explicitly attached by the user; continuity preservation is achieved by the @channelSerial@s for each channel being stored in the recovery key.
+** @(RTN16f)@ When a library is instantiated with the @recover@ client option, it should initialize its internal @msgSerial@ counter to the @msgSerial@ component of the @recoveryKey@. (If the recover fails, the counter should be reset to 0 per "RTN15c7":#RTN15c7 )
+** @(RTN16j)@ When a library is instantiated with the @recover@ client option, for every channel/channelSerial pair in the @recoveryKey@, it should instantiate a corresponding channel and set its "RTL15b":#RTL15b @channelSerial@.
+** @(RTN16k)@ When the library first connects to Ably after being instantiated with a @recover@ client option, it should add an additional @recover@ querystring param to the websocket request, set from the @connectionKey@ component of the @recoveryKey@. Once the library has successfully connected to Ably, it should never again supply a @recover@ querystring param.
+** @(RTN16g)@ @Connection#getRecoveryKey@ is function that returns a string which incorporates the @connectionKey@, the current @msgSerial@, and a collection of pairs of channel @name@ and current @channelSerial@ for every currently attached channel. This must be serialized in a way which is able to encode any unicode channel name. The SDK may assume that the recovery key will only be consumed by the same type of SDK, so this spec does not specify any particular serialization; however, the format should be forward-compatible through the same major version of the SDK.
```

Comment:
I don't think that `get` is a verb that we should be using as a prefix for any method names, especially when those methods are designed to do anything more complex than a 'simple property get'. This observation is made even more stark in this particular context, given it's replacing the previous field/property called `recoveryKey`.

My preference would be to rename this method to `createRecoveryKey`. I think this is more representative of the fact that a call to this method is, as I understand it, capturing a snapshot of state from this Realtime client instance... in that it's likely that a subsequent call to this method will return a different key, on the assumption it's being called while `CONNECTED`.

To further expand on why `get` is a difficult verb to use as a method name prefix for a method which is designed to be available across multiple platforms, in Java it's is idiomatic and common to expect that methods with this prefix are property getters - often returning the value of a `final`, `private` field (and Kotlin only goes on to further expand on that convention).

Reply:
ok, https://github.com/ably/specification/pull/88/commits/ec268811dbf7054e318bf2c92f5e882247675195 changes to `create`

Reply:
https://github.com/ably/ably-dotnet/pull/1185
https://github.com/ably/ably-ruby/pull/387/commits/19a7191a426e8071b95de3f7c40f9f3c1d591242
https://github.com/ably/ably-cocoa/pull/1496/commits/fa251382f46d9f96e34fd930bdf9192b8ff3e9e1
https://github.com/ably/ably-java/pull/842/commits/90dd27588b07f5f324b2e8ea8063f059b28c58f6
https://github.com/ably/ably-js/pull/1045/commits/9c0afb3e5078501c4e6e0c1ebcdbe1e6f0f78650